state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -3207,47 +3187,47 @@ impl<T: Config> AffiliateRegistry for Pallet<T> {
 }
 
 pub mod utilities {
-	use super::*;
+	use cf_amm::math::{invert_price, output_amount_floor, price_from_usd};
+	use sp_runtime::SaturatedConversion;
 
-	/// The amount of a non-gas asset to be used for transaction fee estimation.
-	///
-	/// This should be of a similar order of magnitude to expected fees to get an accurate result.
-	///
-	/// The value should be large enough to allow a good estimation of the fee, but small enough
-	/// to not exhaust the pool liquidity.
-	pub fn estimated_20usd_input(asset: Asset) -> u128 {
-		use cf_primitives::FLIPPERINOS_PER_FLIP;
+	use super::*;
 
+	// Provides a static price that can be used as a fallback when estimating fees/gas.
+	// These prices are rough approximations of real market prices and should be updated
+	// occasionally if the prices move significantly.
+	pub fn hard_coded_price_for_asset(asset: Asset) -> Price {
+		use cf_primitives::FLIP_DECIMALS;
 		const ETH_DECIMALS: u32 = 18;
 		const DOT_DECIMALS: u32 = 10;
 		const BTC_DECIMALS: u32 = 8;
 		const SOL_DECIMALS: u32 = 9;
 
-		/// ~20 Dollars.
-		const FLIP_ESTIMATION_CAP: u128 = 50 * FLIPPERINOS_PER_FLIP;
-		const USD_ESTIMATION_CAP: u128 = 20_000_000;
-		const ETH_ESTIMATION_CAP: u128 = 5 * 10u128.pow(ETH_DECIMALS - 3);
-		const DOT_ESTIMATION_CAP: u128 = 7 * 10u128.pow(DOT_DECIMALS);
-		const BTC_ESTIMATION_CAP: u128 = 2 * 10u128.pow(BTC_DECIMALS - 4);
-		const SOL_ESTIMATION_CAP: u128 = 10 * 10u128.pow(SOL_DECIMALS - 2);
-
 		match asset {
-			Asset::Flip => FLIP_ESTIMATION_CAP,
-			Asset::Usdc => USD_ESTIMATION_CAP,
-			Asset::Usdt => USD_ESTIMATION_CAP,
-			Asset::ArbUsdc => USD_ESTIMATION_CAP,
-			Asset::SolUsdc => USD_ESTIMATION_CAP,
-			Asset::Eth => ETH_ESTIMATION_CAP,
-			Asset::Dot => DOT_ESTIMATION_CAP,
-			Asset::ArbEth => ETH_ESTIMATION_CAP,
-			Asset::Btc => BTC_ESTIMATION_CAP,
-			Asset::Sol => SOL_ESTIMATION_CAP,
-			Asset::HubDot => DOT_ESTIMATION_CAP,
-			Asset::HubUsdc => USD_ESTIMATION_CAP,
-			Asset::HubUsdt => USD_ESTIMATION_CAP,
+			Asset::Usdc |
+			Asset::Usdt |
+			Asset::ArbUsdc |
+			Asset::SolUsdc |
+			Asset::HubUsdc |
+			Asset::HubUsdt => U256::one() << PRICE_FRACTIONAL_BITS, // $1
+			Asset::Flip => price_from_usd(4, FLIP_DECIMALS) / 10, // ~$0.40
+			Asset::Eth | Asset::ArbEth => price_from_usd(2_800, ETH_DECIMALS), // ~$2,800
+			Asset::Dot | Asset::HubDot => price_from_usd(2, DOT_DECIMALS), // ~$2
+			Asset::Btc => price_from_usd(86_500, BTC_DECIMALS),   // ~$86,500
+			Asset::Sol => price_from_usd(127, SOL_DECIMALS),      // ~$127
 		}
 	}
 
+	/// The amount of a non-gas asset to be used for transaction fee estimation.
```

Comment:
This says where it is used, but not what it actually does... Presumably this gives you the amount of asset you would get if you pay 20 USD?

(I know you just copied the comment, but worth updating.)